### ISSUE TITLE
building without polluting the (user-) site-packages

### ISSUE THREAD

**Issue by luziferius** [stele95/LinVAM#28]
You typically build by creating a virtual environment first. So:

``` sh
$ python -m venv venv
$ source venv/bin/activate
$ pip install nuitka
$ run_build_steps
```

On many distributions, pip will refuse to install stuff, as that may break other packages. Running the build in a virtual environment keeps everything clean.

**Comment by stele95** [comment]
In the [Build improvements PR](https://github.com/stele95/LinVAM/pull/30), I've added two ways of installing ``LinVAM`` and both shouldn't pollute user's site-packages.

**Comment by luziferius** [comment]
These steps are for building and _packaging_ the source into installable apps, via Python Wheels, or compiling down to ELF binaries via nuitka. (I have no experience with the latter)
Keeping the virtual environment within the git clone is perfectly fine for this purpose.

I see you implemented something for installing, too.

And in doing so, you basically re-invented [pipx](https://pypi.org/project/pipx/), which is a tool for managing installations of python applications using virtual environments as a backend.

**Comment by stele95** [comment]
Interesting. I have a very little experience with python and developing python apps (basically had none before working on this). Looks like I'll have to investigate a bit more how everything works regarding building and installing python apps, and update this project. Thank you for the suggestions.

**Comment by luziferius** [comment]
When researching, limit it to basically stuff from the last 3 years or so.

The all-new, recommended way of building apps changed relatively drastically, from a setup script called `setup.py` doing everything via [setuptools](https://pypi.org/project/setuptools/) and stuffing everything into a single function call `setup(<huge list of parameters here>)`, towards declaring intent in [`pyproject.toml`](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/) and having multiple build tools reading from there and creating packages for installation.

But I looked around the repository, and it seems it always relied on nuitka to compile down to ELFs, which makes the repository kind of an odd-ball.

**Comment by stele95** [comment]
Thanks for all the info! 

When I first forked this project, it wasn't event set up for building and installing, it expected to be run from ``main.py`` file inside the project's directory. After a quick research, I ran into nuitka, gave it a try and saw that it gets the job done for what I had in mind, so I focused on updating the project and it's features before actually investing some time in a proper way to build it. Maybe that was a mistake? 

Anyway, as soon as I'm done with all the improvements I have in mind for the actual app itself, I'll look into proper building steps and procedures.
